### PR TITLE
TW-746: support get and store fileEntity in database

### PR DESCRIPTION
**Pull request by sherlockvn**

### CODE REVIEW

File: lib/src/utils/models/file_info.dart [hunk 1]
```diff
@@ -5,13 +5,22 @@ import 'package:equatable/equatable.dart';
...
 import 'package:mime/mime.dart';
 
+typedef ProgressCallback = void Function(int count, int total);
+
 class FileInfo with EquatableMixin {
   final String fileName;
   final String filePath;
   final int fileSize;
   final Stream<List<int>>? readStream;
+  final ProgressCallback? progressCallback;
```
Comment (hoangdat) [inline lib/src/utils/models/file_info.dart:15]: what kind of progress? @sherlockvn

Comment (hoangdat) [reply lib/src/utils/models/file_info.dart:15]: and why FileInfo need to have the progress info?

Comment (sherlockvn) [reply lib/src/utils/models/file_info.dart:15]: yeah, i think it better to be inside mixin, so that later upload/download can have progress

Comment (sherlockvn) [reply lib/src/utils/models/file_info.dart:15]: it's progress for upload/download file

Comment (sherlockvn) [reply lib/src/utils/models/file_info.dart]: remove it

File: lib/src/database/hive_collections_database.dart [hunk 2]
```diff
@@ -461,6 +462,11 @@ class HiveCollectionsDatabase extends DatabaseApi {
     return null;
   }
 
+  @override
+  Future<File?> getFileEntity(Uri mxcUri) async {
```
Comment (nqhhdev) [inline lib/src/database/hive_collections_database.dart:466]: Why we need `async`

**Review by nqhhdev — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**